tinygrad/runtime/ops_metal.py

```diff
@@ -44,7 +44,8 @@ def compile_metal(prg, use_xcode=bool(getenv("METAL_XCODE"))) -> bytes:
     # NOTE: if you run llvm-dis on "air" you can see the llvm bytecode
     air = subprocess.check_output(['xcrun', '-sdk', 'macosx', 'metal', '-x', 'metal', '-c', '-', '-o', '-'], input=prg.encode('utf-8'))
     return subprocess.check_output(['xcrun', '-sdk', 'macosx', 'metallib', '-', '-o', '-'], input=air)
-  options = Metal.MTLCompileOptions.alloc().init()
+  options = Metal.MTLCompileOptions.new()
+  options.setFastMathEnabled_(False)
```

Comment:
How much impact does this have to speed?

Reply:
Ok. Not sure. I will benchmark it.

Reply:
I run `external_model_benchmark` on two codes. The results are not different. 

On duyle-py:master:
```
➜  tinygrad git:(0412dafc) ✗ python3 test/external/external_model_benchmark.py                
resnet50        tinygrad_metal_jitless      74.07 ms
resnet50        tinygrad_metal_jit          28.16 ms
resnet50        tinygrad_clang_jitless     945.40 ms
resnet50        tinygrad_clang_jit         877.01 ms
resnet50        torch_cpu                   71.40 ms
resnet50        torch_mps                   28.65 ms
resnet50        onnxruntime_cpu             41.62 ms
resnet50        outputs validated with rtol=2.0e-03, atol=2.0e-03
openpilot       tinygrad_metal_jitless     214.70 ms
openpilot       tinygrad_metal_jit           7.18 ms
openpilot       tinygrad_clang_jitless     275.78 ms
openpilot       tinygrad_clang_jit         156.68 ms
openpilot       onnx2torch              RuntimeError
openpilot       onnxruntime_cpu             46.40 ms
openpilot       outputs validated with rtol=2.0e-03, atol=2.0e-03
efficientnet    tinygrad_metal_jitless     270.41 ms
efficientnet    tinygrad_metal_jit          22.65 ms
efficientnet    tinygrad_clang_jitless     594.11 ms
efficientnet    tinygrad_clang_jit         330.55 ms
efficientnet    torch_cpu                  822.59 ms
efficientnet    torch_mps                   19.29 ms
efficientnet    onnxruntime_cpu             60.48 ms
efficientnet    outputs validated with rtol=2.0e-03, atol=2.0e-03
shufflenet      tinygrad_metal_jitless      57.19 ms
shufflenet      tinygrad_metal_jit           3.82 ms
shufflenet      tinygrad_clang_jitless      78.32 ms
shufflenet      tinygrad_clang_jit          27.71 ms
shufflenet      torch_cpu                   34.73 ms
shufflenet      torch_mps                   73.22 ms
shufflenet      onnxruntime_cpu              5.39 ms
shufflenet      outputs validated with rtol=2.0e-03, atol=2.0e-03
commavq         tinygrad_metal_jitless     207.90 ms
commavq         tinygrad_metal_jit          20.38 ms
commavq         tinygrad_clang_jitless     444.61 ms
commavq         tinygrad_clang_jit         252.42 ms
commavq         onnx2torch       NotImplementedError
commavq         onnxruntime_cpu             78.59 ms
commavq         outputs validated with rtol=2.0e-03, atol=2.0e-03
```
<img width="626" alt="image" src="https://github.com/tinygrad/tinygrad/assets/71242963/cf508b3c-a927-419c-89f1-36a321d2a3a7">

On tinygrad:master: 
```
➜  tinygrad git:(0c9b4ab8) ✗ python3 test/external/external_model_benchmark.py        
resnet50        tinygrad_metal_jitless      76.24 ms
resnet50        tinygrad_metal_jit          28.18 ms
resnet50        tinygrad_clang_jitless     931.83 ms
resnet50        tinygrad_clang_jit         869.03 ms
resnet50        torch_cpu                   74.77 ms
resnet50        torch_mps                   27.04 ms
resnet50        onnxruntime_cpu             39.73 ms
resnet50        outputs validated with rtol=2.0e-03, atol=2.0e-03
openpilot       tinygrad_metal_jitless     206.38 ms
openpilot       tinygrad_metal_jit           6.95 ms
openpilot       tinygrad_clang_jitless     262.44 ms
openpilot       tinygrad_clang_jit         155.86 ms
openpilot       onnx2torch              RuntimeError
openpilot       onnxruntime_cpu             46.52 ms
openpilot       outputs validated with rtol=2.0e-03, atol=2.0e-03
efficientnet    tinygrad_metal_jitless     273.18 ms
efficientnet    tinygrad_metal_jit          22.61 ms
efficientnet    tinygrad_clang_jitless     581.97 ms
efficientnet    tinygrad_clang_jit         328.71 ms
efficientnet    torch_cpu                  805.58 ms
efficientnet    torch_mps                   18.33 ms
efficientnet    onnxruntime_cpu             60.14 ms
efficientnet    outputs validated with rtol=2.0e-03, atol=2.0e-03
shufflenet      tinygrad_metal_jitless      57.67 ms
shufflenet      tinygrad_metal_jit           4.50 ms
shufflenet      tinygrad_clang_jitless      77.96 ms
shufflenet      tinygrad_clang_jit          27.68 ms
shufflenet      torch_cpu                   34.68 ms
shufflenet      torch_mps                   74.99 ms
shufflenet      onnxruntime_cpu              5.39 ms
shufflenet      outputs validated with rtol=2.0e-03, atol=2.0e-03
commavq         tinygrad_metal_jitless     209.29 ms
commavq         tinygrad_metal_jit          19.11 ms
commavq         tinygrad_clang_jitless     445.21 ms
commavq         tinygrad_clang_jit         250.85 ms
commavq         onnx2torch       NotImplementedError
commavq         onnxruntime_cpu             76.97 ms
commavq         outputs validated with rtol=2.0e-03, atol=2.0e-03
```
<img width="638" alt="image" src="https://github.com/tinygrad/tinygrad/assets/71242963/fd72dd64-da99-419e-b024-753a9d52e7b4">